engine/script/src/script_types.cpp

```diff
@@ -0,0 +1,159 @@
+// Copyright 2020-2024 The Defold Foundation
+// Copyright 2014-2020 King
+// Copyright 2009-2014 Ragnar Svensson, Christian Murray
+// Licensed under the Defold License version 1.0 (the "License"); you may not use
+// this file except in compliance with the License.
+//
+// You may obtain a copy of the License, together with FAQs at
+// https://www.defold.com/license
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#include <extension/extension.h>
+#include "script.h"
+
+extern "C"
+{
+#include <lua/lua.h>
+#include <lua/lauxlib.h>
+}
+
+#define LIB_NAME "types"
+
+namespace dmScript
+{
+    /*# Types API documentation
+     *
+     * Functions for checking Defold userdata types.
+     *
+     * @document
+     * @name Types
+     * @namespace types
```

Comment:
pay attention to lua.clj file 
https://github.com/defold/defold/blob/de8e81e893ced421e2a820eeeec70d78675f6212/editor/src/clj/editor/lua.clj#L32-L38

and

https://github.com/defold/defold/blob/de8e81e893ced421e2a820eeeec70d78675f6212/engine/docs/src/script_doc.py#L497-L518